src/Sylius/Behat/spec/Context/Ui/ProductContextSpec.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Behat\Context\Ui;
+
+use Behat\Behat\Context\Context;
+use PhpSpec\ObjectBehavior;
+use Sylius\Behat\Page\Admin\Product\IndexPageInterface;
+use Sylius\Behat\Page\Admin\Product\ShowPageInterface;
+use Sylius\Behat\Page\Product\ProductShowPage;
+use Sylius\Behat\Page\Admin\Product\ShowPage as AdminProductShowPage;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Test\Services\SharedStorage;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class ProductContextSpec extends ObjectBehavior
+{
+    public function let(
+        SharedStorageInterface $sharedStorage,
+        ProductShowPage $productShowPage,
+        ShowPageInterface $adminProductShowPage,
+        IndexPageInterface $productIndexPage
+    ) {
+        $this->beConstructedWith($sharedStorage, $productShowPage, $adminProductShowPage, $productIndexPage);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Behat\Context\Ui\ProductContext');
+    }
+
+    function it_implements_context_interface()
+    {
+        $this->shouldImplement(Context::class);
+    }
+
+    function it_checks_if_i_am_able_to_access_product_page($productShowPage, ProductInterface $product)
+    {
+        $productShowPage->tryToOpen(['product' => $product])->shouldBeCalled();
+        $productShowPage->isOpen(['product' => $product])->willReturn(true);
+        
+        $this->iShouldBeAbleToAccessProduct($product);
+    }
+
+    function it_checks_if_i_am_not_able_to_access_product_page($productShowPage, ProductInterface $product)
+    {
+        $productShowPage->tryToOpen(['product' => $product])->shouldBeCalled();
+        $productShowPage->isOpen(['product' => $product])->willReturn(false);
+
+        $this->iShouldNotBeAbleToAccessProduct($product);
```

Comment:
I'm asking about the case when `$productShowPage->isOpen(['product' => $product])->willReturn(true);` and I will call `$this->iShouldNotBeAbleToAccessProduct($product)`. It will throw exception and there is not spec for that.